definitions/checks/puppet/provide_upgrade_guide.rb

```diff
@@ -0,0 +1,34 @@
+module Checks::Puppet
+  class ProvideUpgradeGuide < ForemanMaintain::Check
+    metadata do
+      description 'Verify puppet and provide upgrade guide for it'
+
+      confine do
+        feature(:downstream) &&
+          feature(:downstream).current_minor_version == '6.3' &&
+          find_package('puppet')
+      end
+
+      manual_detection
+    end
+
+    def run
+      with_spinner('Verifying puppet version before upgrade') do |spinner|
+        puppet_package = find_package('puppet')
+        spinner.update "current puppet version: #{puppet_package}"
+        if puppet_package =~ /puppet-3/
+          fail!(<<-MESSAGE.strip_heredoc)
```

Comment:
Shouldn't we extend https://github.com/theforeman/foreman_maintain/blob/master/definitions/procedures/knowledge_base_article.rb to support docs as well as kcs and use it here?

Reply:
@iNecas,
Done with the changes and updated the PR.